Objects/unicodeobject.c

```diff
@@ -3820,24 +3820,33 @@ PyUnicode_AsUTF8AndSize(PyObject *unicode, Py_ssize_t *psize)
 {
     if (!PyUnicode_Check(unicode)) {
         PyErr_BadArgument();
+        *psize = 0;
```

Comment:
`psize` can be NULL. I have added tests for this: #111100.

Reply:
Oh I see. Thanks for additional tests. I moved this unrelated change to its own PR: PR #111106.